src/components/MainTable/MainTable.tsx

```diff
@@ -10,99 +10,111 @@ import TableHeader from "../TableHeader";
 import TableCell from "../TableCell";
 import type { TableCellProps } from "../TableCell";
 
-export type MainTableHeader = {
-  /**
-   * The content of the table header.
-   */
-  content?: ReactNode;
-  /**
-   * Optional classes to apply to the table header.
-   */
-  className?: ClassName;
-  /**
-   * A key to sort the rows by. It should match a key given to the row `sortData`.
-   */
-  sortKey?: string | null;
-} & Omit<HTMLProps<HTMLTableHeaderCellElement>, "content">;
+export type MainTableHeader = PropsWithSpread<
+  {
+    /**
+     * The content of the table header.
+     */
+    content?: ReactNode;
+    /**
+     * Optional classes to apply to the table header.
+     */
+    className?: ClassName;
+    /**
+     * A key to sort the rows by. It should match a key given to the row `sortData`.
+     */
+    sortKey?: string | null;
+  },
+  HTMLProps<HTMLTableHeaderCellElement>
+>;
 
-export type MainTableCell = {
-  /**
-   * The content of the table cell.
-   */
-  content?: ReactNode;
-} & Omit<TableCellProps, "children" | "content">;
+export type MainTableCell = PropsWithSpread<
+  {
+    /**
+     * The content of the table cell.
+     */
+    content?: ReactNode;
+  },
+  Omit<TableCellProps, "children">
```

Comment:
Same here, if this still needs to be explicitly omitted then a comment might help.